network/framework/src/protocols/health_checker/mod.rs

```diff
@@ -207,15 +209,19 @@ impl<NetworkClient: NetworkClientInterface<HealthCheckerMsg> + Unpin> HealthChec
                 }
                 conn_event = connection_events.select_next_some() => {
                     match conn_event {
-                        ConnectionNotification::NewPeer(metadata, _network_id) => {
-                            self.network_interface.create_peer_and_health_data(
-                                metadata.remote_peer_id, self.round
-                            );
+                        ConnectionNotification::NewPeer(metadata, network_id) => {
+                            if network_id == self_network_id {
```

Comment:
Can we add a comment above here mentioning why we need to ignore notifications for irrelevant peers?

Ideally, we'd actually be able to subscribe to a specific network ID, so we don't receive these notifications, but I recognize that this is a quicker approach to unblocking the PR 😄 

In the future, we should probably update the health checker to only run a single instance, and not one per network. But, we could also migrate to the peer monitoring service, which may be cleaner.

Reply:
Commented. My rationalization is that PeersAndMetadata is a global singleton across all "networks"